### PR TITLE
Addd Github action to build and test the project

### PR DESCRIPTION
**Pull request by alejandrodnm**
### Add github action to build and test the project 
 
 The workflow will:

- Build the project
- Run the unittest
- Check for lint issues
- Check for format issues

### CODE REVIEW

File: .github/workflows/go.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+
+    - name: Build
+      run: go build
+
+  test:
```
Comment (yurishkuro) [inline .github/workflows/go.yaml:37]: this feels like an overkill for this repo, I would just combine everything in one runner, build, test, lint.

**Review by yurishkuro — COMMENTED** [review]
Thanks. I suggest simplifying the workflow.

File: .github/workflows/go.yaml (hunk 1 above)
Comment (alejandrodnm) [reply .github/workflows/go.yaml]: done :)

**Review by yurishkuro — APPROVED** [review]
🎉

**Pull request merged**